.pre-commit-config.yaml

```diff
@@ -0,0 +1,52 @@
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.2.0
+    hooks:
+      - id: check-yaml
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+  - repo: https://github.com/python/black
+    rev: 22.3.0
```

Comment:
Does pre-commit auto-install hooks? (thanks in advance for dealing with my ignorance about pre-commit!) If not, let's pin the dev dependencies for each of these in setup.py. If it does auto-install dependencies, let's remove those dependencies from setup.py.

And I guess we should add pre-commit itself as a dev dependency in setup.py?

Reply:
Yup, it will install the specified hook at the given rev, so it will install when it first runs.  The great thing is that if you change the rev it will install the new one too.  Since it's all in a virtual environment pre-commit uses, you could remove them as dependencies in `setup.py` if you wanted, and pre-commit will manage it.

Note that this is only true for non-local hooks (local ones are at the end, and those run on the given system so things like `pytest` would still be needed).

Reply:
> Since it's all in a virtual environment pre-commit uses, you could remove them as dependencies in `setup.py` if you wanted, and pre-commit will manage it.

Fantastic; let's do that.